great_expectations_provider/operators/great_expectations.py

```diff
@@ -251,48 +252,97 @@ def make_connection_string(self) -> str:
                 odbc_connector = "mssql+pyodbc"
             uri_string = f"{odbc_connector}://{self.conn.login}:{self.conn.password}@{self.conn.host}:{self.conn.port}/{self.schema}"  # noqa
         elif conn_type == "snowflake":
-            snowflake_account = (
-                self.conn.extra_dejson.get("account") or self.conn.extra_dejson["extra__snowflake__account"]
-            )
-            snowflake_region = (
-                self.conn.extra_dejson.get("region") or self.conn.extra_dejson["extra__snowflake__region"]
-            )
-            snowflake_database = (
-                self.conn.extra_dejson.get("database") or self.conn.extra_dejson["extra__snowflake__database"]
-            )
-            snowflake_warehouse = (
-                self.conn.extra_dejson.get("warehouse") or self.conn.extra_dejson["extra__snowflake__warehouse"]
-            )
-            snowflake_role = self.conn.extra_dejson.get("role") or self.conn.extra_dejson["extra__snowflake__role"]
+            try:
+                from airflow.providers.snowflake.hooks.snowflake import SnowflakeHook
+                from cryptography.hazmat.backends import default_backend
+                from cryptography.hazmat.primitives import serialization
+
+                hook = SnowflakeHook(snowflake_conn_id=self.conn_id)
+
+                # Support the operator overriding the schema
+                # which is necessary for temp tables.
+                hook.schema = self.schema or hook.schema
+
+                conn = hook.get_connection(self.conn_id)
```

Comment:
How is this different than the existing `self.conn`? Same with `hook.schema` above -- it's already the same as self.conn.schema if one isn't provided.

Reply:
The hook doesn't know about the operator's `schema` argument that is why it needs to be explicitly set on line 264. Using the hook to get the connection just seemed more in line with the reasoning behind this change, which is to use the Snowflake provider and avoid duplicating the code.